test/test-growpart-overprovision

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+# NEED_ROOT
+
```

Comment:
I don't think there is really any reason that test of this function needs root.
tests that don't require root are much more safely run for a developer and thus more generally useful.

loop devices, but there isn't any real reason for that.
I think a test like test-growpart-fsimage would be just as good.

 * create a disk image
 * partition it
 * grow it
 * read the new partition table
 * verify expected.

Reply:
I agree, it would be nicer not to require root, especially as the test touches block devices (scary), however we're running it in CI, so I don't think it's worth blocking the PR from landing, wdyt?

Cc: @blackboxsw

Reply:
@paride

Sorry for the lack of feedback - I have rewritten the test to not require root. Whilst doing I spotted what appeared to be some off-by-1 references to sectors in the growpart code, fixed those and some of the other tests broke, and I got a little lost in comparing sfdisk/sgdisk/parted output to get my mind straight regarding sectors :-)

Anyway I think I have it cracked now and this weekend expect to do another push to finalise things.

Reply:
Ok. Additional changes pushed. That should be everything addressed.

Reply:
@dermotbradley looks like the code in your PR still requires root, but you mentioned it should not require root now, so I'm a bit confused. Did you push your latest version?

Reply:
@paride Ah, my mistake, I had pushed the wrong version of the file - corrected now.